lib/core/managementClient.ts

```diff
@@ -532,6 +556,209 @@ export class ManagementClient extends LinkEntity {
+
+      log.mgmt("[%s] Get rules request body: %O.", this._context.namespace.connectionId,
+        request.body);
+      log.mgmt("[%s] Acquiring lock to get the management req res link.",
+        this._context.namespace.connectionId);
+      await defaultLock.acquire(this.managementLock, () => { return this._init(); });
+
+      const response = await this._mgmtReqResLink!.sendRequest(request);
+      if (response.application_properties!.statusCode === 204 || !response.body || !Array.isArray(response.body.rules)) {
+        return [];
+      }
+
+      // Reference: https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-amqp-request-response#response-11
+      const result: { "rule-description": Typed }[] = response.body.rules || [];
+      const rules: RuleDescription[] = [];
+      result.forEach((x) => {
+        const ruleDescriptor = x['rule-description'];
+        if (!Array.isArray(ruleDescriptor.value) || ruleDescriptor.value.length < 3) {
+          return;
+        }
+        const filtersRawData: Typed = ruleDescriptor.value[0];
+        const actionsRawData: Typed = ruleDescriptor.value[1];
+        const rule: RuleDescription = {
+          name: ruleDescriptor.value[2].value
+        };
+
+        switch (filtersRawData.descriptor.value) {
+          case 83483426823:
+            rule.filter = {
+              expression: "1=1"
+            };
+            break;
+          case 83483426824:
+            rule.filter = {
+              expression: "1=0"
+            };
+            break;
+          case 83483426822:
+            if (Array.isArray(filtersRawData.value) && filtersRawData.value.length === 2) {
+              rule.filter = {
+                expression: filtersRawData.value[0] && filtersRawData.value[0].value
+              };
+            }
+            break;
+          case 83483426825:
+            if (Array.isArray(filtersRawData.value) && filtersRawData.value.length === 9) {
+              rule.filter = {
+                correlationId: filtersRawData.value[0] && filtersRawData.value[0].value,
+                messageId: filtersRawData.value[1] && filtersRawData.value[1].value,
+                to: filtersRawData.value[2] && filtersRawData.value[2].value,
+                replyTo: filtersRawData.value[3] && filtersRawData.value[3].value,
+                label: filtersRawData.value[4] && filtersRawData.value[4].value,
+                sessionId: filtersRawData.value[5] && filtersRawData.value[5].value,
+                replyToSessionId: filtersRawData.value[6] && filtersRawData.value[6].value,
+                contentType: filtersRawData.value[7] && filtersRawData.value[7].value,
+                userProperties: filtersRawData.value[8] && filtersRawData.value[8].value,
+              };
+            }
+            break;
+          default:
+            break;
+        }
+
+        if (actionsRawData.descriptor.value === 1335734829062
```

Comment:
I think it would help with code readability if we put these magic numbers in constants.

Reply:
As soon as https://github.com/Azure/amqp-common-js/pull/13 gets in :)

Reply:
Done